src/sanitizer.js

```diff
@@ -57,13 +57,38 @@ const _transformConfig = function transformConfig(config) {
   const allowElems = config.allowElements || [];
   const allowAttrs = config.allowAttributes || [];
   const blockElements = config.blockElements || [];
+  const dropElements = config.dropElements || [];
   const dropAttrs = config.dropAttributes || [];
-  return {
-    ALLOWED_TAGS: allowElems,
-    ALLOWED_ATTR: allowAttrs,
-    FORBID_TAGS: blockElements,
-    FORBID_ATTR: dropAttrs,
-  };
+  // https://github.com/cure53/DOMPurify/issues/556
+  const isdropElementsSet =
+    dropElements !== DEFAULT_DROP_ELEMENTS && dropElements !== [];
+  const isblockElementsSet =
+    blockElements !== DEFAULT_BLOCKED_ELEMENTS && blockElements !== [];
+  if (!isblockElementsSet && isdropElementsSet) {
```

Comment:
I think it would be useful to have a couple of comments for the three if-cases here.